config/env_config.go

```diff
@@ -0,0 +1,431 @@
+
+	// The default authentication requirements for this API.
+	Authentication AuthenticationRequirement `yaml:"authentication,omitempty" mapstructure:"authentication,omitempty"`
+
+	// The default consumer authorization requirements for this API.
+	ConsumerAuthorization ConsumerAuthorization `yaml:"consumer_authorization,omitempty" mapstructure:"consumer_authorization,omitempty"`
+
+	// Transformation rules applied to HTTP requests.
+	HTTPRequestTransforms HTTPRequestTransformations `yaml:"http_request_transforms,omitempty" mapstructure:"http_request_transforms,omitempty"`
+
+	// A list of API Operations, names of which must be unique within the API.
+	Operations []APIOperation `yaml:"operations" mapstructure:"operations"`
+}
+
+// An APIOperation associates a set of rules with a set of request matching settings.
+type APIOperation struct {
+	// Name of the API Operation. Unique within a API.
+	Name string `yaml:"name" mapstructure:"name"`
+
+	// The authentication requirements for thie Operation. If specified, this overrides the default AuthenticationRequirement specified at the API level.
+	Authentication AuthenticationRequirement `yaml:"authentication,omitempty" mapstructure:"authentication,omitempty"`
+
+	// The consumer authorization requirement for this Operation. If specified, this overrides the default ConsumerAuthorization specified at the API level.
+	ConsumerAuthorization ConsumerAuthorization `yaml:"consumer_authorization,omitempty" mapstructure:"consumer_authorization,omitempty"`
+
+	// HTTP matching rules for this Operation. If omitted, this API Operation will match all HTTP requests not matched by another API Operation.
+	HTTPMatches []HTTPMatch `yaml:"http_match,omitempty" mapstructure:"http_match,omitempty"`
+
+	// Transformation rules applied to HTTP requests for this Operation. Overrides the rules set at the API level.
+	HTTPRequestTransforms HTTPRequestTransformations `yaml:"http_request_transforms,omitempty" mapstructure:"http_request_transforms,omitempty"`
+}
+
+// HTTPRequestTransformations are rules for modifying HTTP requests.
+type HTTPRequestTransformations struct {
+	// Header values to append. If a specified header is already present in the request, an additional value is added.
+	AppendHeaders map[string]string `yaml:"append_headers,omitempty" mapstructure:"append_headers,omitempty"`
+
+	// Header values to set. If a specified header is already present, the value here will overwrite it.
+	SetHeaders map[string]string `yaml:"set_headers,omitempty" mapstructure:"set_headers,omitempty"`
+
+	// Headers to remove. Supports single wildcard globbing e.g. `x-apigee-*`.
+	RemoveHeaders []string `yaml:"remove_headers,omitempty" mapstructure:"remove_headers,omitempty"`
+}
+
+// AuthenticationRequirement defines the authentication requirement. It can be jwt, any or all.
+type AuthenticationRequirement struct {
+	AuthenticationRequirements AuthenticationRequirements `yaml:"-"`
```

Comment:
Referencing AuthenticationRequirements from an Operation like this will be:  `operation.Authentication.AuthenticationRequirements`. I think it would read better if we just made the field name `Requirements` so the reference name would be: `operation.Authentication.Requirements`.

Reply:
Yes this makes sense. Will do.